kernel/hashlib.h

```diff
@@ -100,6 +106,20 @@ template<typename P, typename Q> struct hash_ops<std::pair<P, Q>> {
 	}
 };
 
+template<typename P, typename Q> struct hash_ops<dict<P, Q>> {
```

Comment:
Template specialisation is only necssary for data structures that we have no control over. Things that we do have control over should provide a `T::hash()` function so that the default template takes over:

https://github.com/YosysHQ/yosys/blob/da16a5b83a71db4f1d15b5ee3d4f6ab6f54fc0d2/kernel/hashlib.h#L59-L66

Reply:
Oh, I see.